### PR TITLE
DF-546 (+ DF-724 & DF-770): BE record creators details page

### PR DESCRIPTION
**Pull request by TNA-Allan**
Ticket URL: 
https://national-archives.atlassian.net/browse/DF-546
https://national-archives.atlassian.net/browse/DF-724
https://national-archives.atlassian.net/browse/DF-770

**Please do not merge until FE is completed** 

## About these changes

A new record details page - Record Creators is introduced in Search

## How to check these changes

1) 
- Navigate to Record creators bucket
- Select the record from the results area
- Filter and select records for each Creator type

2) 
http://127.0.0.1:8000/catalogue/id/F257721/
http://127.0.0.1:8000/catalogue/id/F234763/
http://127.0.0.1:8000/catalogue/id/F125932/

2)  For an archive record

Select records creators from Collection information 
http://127.0.0.1:8000/catalogue/id/A13532479/

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer.
- [x] Checked PR title starts with ticket number as per project conventions to help us keep track of changes.
- [x] Ensured that PR includes only commits relevant to the ticket.
- [x] Waited for all CI jobs to pass before requesting a review.
- [x] Added/updated tests and documentation where relevant.

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

### CODE REVIEW

**Comment by TNA-Allan** [comment]
I’ve set it up for DRAFT mode as after I did BE work, got to know it will require some FE updates as mentioned in the ticket.. 
PR is open to review from BE perspective

File: etna/records/models.py [hunk 1]
```diff
@@ -713,6 +720,188 @@ def archive_accessions(self) -> Optional[AccessionsInfo]:
     def archive_repository_url(self) -> str:
         return self.get("repository.url", "")
 
+    @cached_property
+    def alternative_names(self) -> tuple(dict):
+        """ """
```
Comment (jamesbiggs) [inline etna/records/models.py:725]: Empty doc-string :)

File: etna/records/models.py [hunk 2]
```diff
@@ -713,6 +720,188 @@ def archive_accessions(self) -> Optional[AccessionsInfo]:
...
+        seperator = "-"
+        joiner = "; "
+        person_date = service_activity_date = ""
+        if self.birth_date and self.death_date:
+            person_date = f"{self.birth_date}{seperator}{self.death_date}"
+        else:
+            person_date = self.birth_date or self.death_date
+
+        if self.start_date and self.end_date:
+            service_activity_date = f"{self.start_date}{seperator}{self.end_date}"
```
Comment (jamesbiggs) [inline etna/records/models.py:877]:
```suggestion
        separator = "-"
        joiner = "; "
        person_date = service_activity_date = ""
        if self.birth_date and self.death_date:
            person_date = f"{self.birth_date}{separator}{self.death_date}"
        else:
            person_date = self.birth_date or self.death_date

        if self.start_date and self.end_date:
            service_activity_date = f"{self.start_date}{separator}{self.end_date}"
```

Just a small spelling change :)

**Review by jamesbiggs — COMMENTED** [review]
Hi Allan, nice work 😄 Just two small changes here then this can be approved 👌

File: etna/records/models.py (hunk 1 above)
Comment (TNA-Allan) [reply etna/records/models.py]: Fixed. Thanks :)

File: etna/records/models.py (hunk 2 above)
Comment (TNA-Allan) [reply etna/records/models.py]: Fixed. Thanks :)

**Review by jamesbiggs — APPROVED** [review]
Approved BE 👍 Nice work 😄

**Comment by matt-blair** [comment]
@TNA-Allan I've just made the requested FE changes:

- Removed the reference to Summary of Collections from the main Record Creators template
- Amended the Alternative Names description field to a <p> tag and created an associated style
- Reduced the top margin spacing for the h1 heading

If you are able to double check those changes that would be great.

**Comment by TNA-Allan** [comment]
> @TNA-Allan I've just made the requested FE changes:
> 
> * Removed the reference to Summary of Collections from the main Record Creators template
> * Amended the Alternative Names description field to a p tag and created an associated style
> * Reduced the top margin spacing for the h1 heading
> 
> If you are able to double check those changes that would be great.

Thanks @matt-blair 

Looks good to me as long as its disconnected from class for related records.

[class="details-related-records__list"](https://github.com/nationalarchives/ds-wagtail/blob/feature/DF-546-be-record-creators-details-page/templates/includes/records/record_creator_description.html#L71-L73) for `{{ place }}` Maybe this is missed, please can you check/update?

I've put a comment in the ticket for FE PR to review.

**Comment by JohnHeeryTNA** [comment]
Hi @matt-blair 
I think we have a layout bug on smaller viewports for an archive record.
If there is a long url for the Website entry, it seems to break out of the container on smaller viewports - 768px and below

Screenshot 2023-07-12 at 12 04 47

**Comment by matt-blair** [comment]
@JohnHeeryTNA Thanks for spotting this. I've just add a word-wrap fix on the link specifically to ensure these links don't cause any layout issues. Are you ok to double-check it?

@TNA-Allan As discussed with you and Sam, I've also incorporated DF-724 (removing the 'view on a map' link) and DF-770 (Adding back to search / start a new search) to this PR to fast-track these through as part of this ticket. Links to the tickets are in the description above.

**Comment by JohnHeeryTNA** [comment]
FE Change looks good @matt-blair

**Review by JohnHeeryTNA — APPROVED** [review]
approving FE changes

**Pull request merged**